privacyidea/lib/policymatch.py

```diff
@@ -0,0 +1,175 @@
+    code should use one of the provided classmethods to construct a ``Match`` object. See the respective
+    classmethods for details.
+
+    A ``Match`` object encapsulates a policy matching operation, i.e. a call to ``PolicyClass.match_policies``.
+    In order to retrieve the matching policies, one should use one of ``policies()``, ``action_values()`` and ``any()``.
+    """
+    def __init__(self, g, **kwargs):
+        self._g = g
+        self._match_kwargs = kwargs
+
+    def policies(self, write_to_audit_log=True):
+        """
+        Return a list of policies, sorted by priority (prioritized policies appear first).
+        :param write_to_audit_log: If True, write the list of matching policies to the audit log
+        :return: a list of policy dictionaries
+        """
+        if write_to_audit_log:
+            audit_data = self._g.audit_object.audit_data
+        else:
+            audit_data = None
+        return self._g.policy_object.match_policies(audit_data=audit_data,
+                                                    **self._match_kwargs)
+
+    def any(self, write_to_audit_log=True):
+        """
+        Return True if at least one policy matches.
+        :param write_to_audit_log: If True, write the list of matching policies to the audit log
+        :return: True or False
+        """
+        return bool(self.policies(write_to_audit_log=write_to_audit_log))
+
+    def action_values(self, unique, allow_white_space_in_action=False, write_to_audit_log=True):
+        """
+        Return a dictionary of action values of the matching policies.
+        :param unique: If True, return only the prioritized action value.
+                       See ``PolicyClass.get_action_values`` for details.
+        :param allow_white_space_in_action: If True, allow whitespace in action values.
+                       See ``PolicyClass.get_action_values`` for details.
+        :param write_to_audit_log: If True, augment the audit log with the names of all
+                       policies whose action values are returned
+        :return:
+        """
+        policies = self.policies(write_to_audit_log=False)
+        action_values = self._g.policy_object.extract_action_values(policies,
+                                                                    self._match_kwargs['action'],
+                                                                    unique=unique,
+                                                                    allow_white_space_in_action=
+                                                                    allow_white_space_in_action)
+        if write_to_audit_log:
+            for action_value, policy_names in action_values.items():
+                for p_name in policy_names:
+                    self._g.audit_object.audit_data.setdefault("policies", []).append(p_name)
+        return action_values
+
+    @classmethod
+    def simple(cls, g, scope, action, realm, user):
```

Comment:
as realm and user are exclusive pleases split this in two methods *user* and *realm*.

Reply:
done in a11ccd0d1a2a1d94e5c2be6a02fae4454f636279.
I also added a classmethod ``action_only``, which is also used in some places.